utils/src/ast-grep/resolve-binding-path.test.ts

```diff
@@ -0,0 +1,159 @@
+import assert from "node:assert/strict";
+import { describe, it } from "node:test";
+import astGrep from "@ast-grep/napi";
+import dedent from "dedent";
+
+import { resolveBindingPath } from "./resolve-binding-path.ts";
+
+describe("resolve-binding-path", () => {
+	it("should be able to solve binding path to simple requires", () => {
+		const code = dedent`
+			const util = require('node:util');
+		`;
+
+		const rootNode = astGrep.parse(astGrep.Lang.JavaScript, code);
+		const importStatement = rootNode.root().find({
+			rule: {
+				kind: "variable_declarator",
+			},
+		});
+
+		const bindingPath = resolveBindingPath(importStatement!, "$.types.isNativeError");
+
+		assert.strictEqual(bindingPath, "util.types.isNativeError");
+	});
+
+	it("should be able to solve binding path when destructuring happen", () => {
+		const code = dedent`
+			const { types } = require('node:util');
+		`;
+
+		const rootNode = astGrep.parse(astGrep.Lang.JavaScript, code);
+		const importStatement = rootNode.root().find({
+			rule: {
+				kind: "variable_declarator",
+			},
+		});
+
+		const bindingPath = resolveBindingPath(importStatement!, "$.types.isNativeError");
+
+		assert.strictEqual(bindingPath, "types.isNativeError");
+	});
+
+	it("should be able to solve binding when have multiple destructurings", () => {
+		const code = dedent`
+			const { types: { isNativeError } } = require('node:util');
+		`;
+
+		const rootNode = astGrep.parse(astGrep.Lang.JavaScript, code);
+		const importStatement = rootNode.root().find({
+			rule: {
+				kind: "variable_declarator",
+			},
+		});
+
+		const bindingPath = resolveBindingPath(importStatement!, "$.types.isNativeError");
+
+		assert.strictEqual(bindingPath, "isNativeError");
+	});
+
+	it("should be able to solve binding when a rename happen", () => {
+		const code = dedent`
+			const { types: typesRenamed } = require('node:util');
+		`;
+
+		const rootNode = astGrep.parse(astGrep.Lang.JavaScript, code);
+		const importStatement = rootNode.root().find({
+			rule: {
+				kind: "variable_declarator",
+			},
+		});
+
+		const bindingPath = resolveBindingPath(importStatement!, "$.types.isNativeError");
+
+		assert.strictEqual(bindingPath, "typesRenamed.isNativeError");
+	});
+
+	it("should throw an error if unsupported node kind is passed", () => {
+		const code = dedent`
+			function test() { }
+		`;
+
+		const rootNode = astGrep.parse(astGrep.Lang.JavaScript, code);
+		const functionDeclaration = rootNode.root().find({
+			rule: {
+				kind: "function_declaration",
+			},
+		});
+
+		assert.throws(() => resolveBindingPath(functionDeclaration!, "$.types.isNativeError"));
```

Comment:
why not just putting null and handling at codemod level ?

The question is deeper that just here. Should our utility throw error or should return null and handled in codemod ?

My tought is throw error but document it so we can catch it as codemod level

Reply:
My idea behind throwing an error was to make development easier, to alert whoever is working on it that the node being passed is incorrect in some way. But yeah, I agree with you, it does make the usage a bit awkward.

I'll go ahead and remove those throw scenarios.

Reply:
Let's keep it !